README.md

```diff
@@ -1,28 +1,157 @@
+oc process -f https://raw.githubusercontent.com/jaegertracing/jaeger-openshift/master/all-in-one/jaeger-all-in-one-template.yml | oc create -f -
 ```
 
-## Troubleshooting
-Tracer sometimes fails to resolve Jaeger's address. In such a case run:
+Once everything is ready, the Jaeger Query UI can be accessed via https://jaeger-query-jaeger.127.0.0.1.nip.io/ .
+
+## Production setup
+
+### TL;DR
+
+Install with:
 ```bash
-sudo iptables -F
+oc login -u system:admin
+oc new-project jaeger-infra
+oc process -f https://raw.githubusercontent.com/jaegertracing/jaeger-openshift/master/production/jaeger-production-template.yml | oc create -f -
+oc policy add-role-to-user cluster-admin developer -n jaeger-infra
+```
+
+Once everything is ready, the Jaeger Query UI can be accessed via https://jaeger-query-jaeger.127.0.0.1.nip.io/ .
+
+Your Agent hostname is `jaeger-agent.jaeger-infra.svc.cluster.local`
+
+Clean up with:
+```bash
+oc delete all,template,daemonset -l jaeger-infra
+```
+
+### Detailed instructions
+
+You may choose to deploy the Jaeger instance on its own namespace or on an existing namespace,
+with the installation done by an admin or by an unprivileged user. In any case, you'll need your
+Tracer to be pointed to the Agent, which is available at a hostname following this pattern:
+
+```
+jaeger-agent.${NAMESPACE}.svc.cluster.local
+```
+
+For instance, when deployed on its own namespace `jaeger-infra`, this is the full hostname:
+
+```
+jaeger-agent.jaeger-infra.svc.cluster.local
+```
+
+#### Creating the deployment
+
+The `jaeger-production-template.yml` file provides a quick start to deploy a production-ready instance of Jaeger on OpenShift.
+The template includes a `StatefulSet` for Cassandra, a `DaemonSet` for the Agent, a `Job` that creates the schema,
+as well as regular services for the individual components such as `Collector` and `Query`. Each one of those
+can be managed and scaled individually.
+
+#### Installing Jaeger on its own namespace as `admin`
+
+Ideally, the Jaeger components would run on its own namespace. For the recommended setup, these are the commands
+to be issued:
+
+```bash
+oc login -u system:admin
+oc new-project jaeger
+oc process -f jaeger-production-template.yml | oc create -f -
+```
+
+To be able to manage the objects via the UI, add the appropriate permissions to your user. For development
+purposes, this can be executed to get the `developer` user to access the Jaeger namespace as `cluster-admin`:
+
+```bash
+oc login -u system:admin
+oc policy add-role-to-user cluster-admin developer -n jaeger
+```
+
+#### Installing Jaeger under an existing namespace as `developer`
+
+Jaeger can also be installed as `developer` (or any other user). As the `DaemonSet` runs at the node level,
+however, your user need to have such permission. It can be achieved by running this:
+
+```bash
+oc login -u system:admin
+oc create -f daemonset-admin.yml
+oadm policy add-role-to-user daemonset-admin developer
```

Comment:
I'll keep this as `oadm` for the moment, as the switch from `oadm` to `oc adm` is relatively recent, so, it might confuse people using older `oc` versions. But that's a good catch!